cmd/nodecmd/wiz.go

```diff
@@ -1001,3 +1001,94 @@ func setUpSubnetLogging(clusterName, subnetName string) error {
 	spinSession.Stop()
 	return nil
 }
+
+func addBlockchainToRelayerConf(network models.Network, cloudNodeID string, blockchainName string) error {
+	relayerAddress, relayerPrivateKey, err := teleporter.GetRelayerKeyInfo(app.GetKeyPath(constants.AWMRelayerKeyName))
+	if err != nil {
+		return err
+	}
+
+	storageBasePath := constants.AWMRelayerDockerDir
+	configBasePath := app.GetNodeInstanceDirPath(cloudNodeID)
+
+	configPath := app.GetAWMRelayerServiceConfigPath(configBasePath)
+	if err := os.MkdirAll(filepath.Dir(configPath), constants.DefaultPerms755); err != nil {
+		return err
+	}
+	ux.Logger.PrintToUser("updating configuration file %s", configPath)
+
+	if err := teleporter.CreateBaseRelayerConfigIfMissing(
+		configPath,
+		logging.Info.LowerString(),
+		app.GetAWMRelayerServiceStorageDir(storageBasePath),
+		network,
+	); err != nil {
+		return err
+	}
+
+	chainSpec := contract.ChainSpec{CChain: true}
+	subnetID, err := contract.GetSubnetID(app, network, chainSpec)
+	if err != nil {
+		return err
+	}
+	blockchainID, err := contract.GetBlockchainID(app, network, chainSpec)
+	if err != nil {
+		return err
+	}
+	registryAddress, messengerAddress, err := contract.GetICMInfo(app, network, chainSpec, false, false, false)
+	if err != nil {
+		return err
+	}
+	rpcEndpoint, wsEndpoint, err := contract.GetBlockchainEndpoints(app, network, chainSpec, false, false)
+	if err != nil {
+		return err
+	}
+
+	if err = teleporter.AddSourceAndDestinationToRelayerConfig(
+		configPath,
+		rpcEndpoint,
+		wsEndpoint,
+		subnetID.String(),
+		blockchainID.String(),
+		registryAddress,
+		messengerAddress,
+		relayerAddress,
+		relayerPrivateKey,
+	); err != nil {
+		return err
+	}
+
+	chainSpec = contract.ChainSpec{BlockchainName: blockchainName}
```

Comment:
why twice to do the same thing ?

Reply:
it is not the same, one is done for cchain, the other one for the currently deployed blockchain